spec: Propose specification: Location

mikaelbr opened the pull request:
# Proposal: Location

Retrieve location data periodically.

Fixes #2

## Context and known limitations

Specified as a subset of [Ruter Location](https://github.com/RuterNo/adt-doc/blob/master/docs/v2.6.0/json-schemas/location.json) but with local MQTT (no specifying operator and vehicle). Mainly support for lat, lon, and date.

Has no configuration like interval or format.

---

## Checklist

- [x] Includes documentation file with description, topics, and examples.
- [x] Includes JSON Schema
- [x] Includes passing and failing tests.

## Confirmed and verified by parties

- [x] MT Buss
- [ ] Consat
- [x] Fara

mikaelbr committed:
spec: propose specification Location

mikaelbr commented:
- Add fields about accuracy and ability to say something about quality (additional fields).
- Information on movement.


---

Check with Consat on complete Ruter location JSON Schema

mikaelbr commented:
Add note on standard interval. 1 per sec.

mikaelbr committed:
updates location spec according to feedback

mikaelbr committed:
updates topic / schema path in docs

mikaelbr committed:
docs: adds topic tree

Topguy commented on specifications/sensors/gnss/location.json:
```diff
@@ -0,0 +1,83 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://github.com/mrfylke/mtbuss-mqtt-validator/specifications/location.json",
```
Full path har vel blitt  /specifications/sensors/gnss/location.json  ?

mikaelbr replied:
Stemmer. Tenkte å restrukturere litt paths i dag for å samle dokumentasjon og schemas på samme sted.

mikaelbr committed:
refactor: changes location to be in line with ruter v3

mikaelbr commented:
**Information**:
Ruter has an unmerged breaking change in how to handle Location in what they call adt v3. To be future proof and match Ruter I've updated this proposal to match Ruters _new_ Location specification.

Primary changes:
- Added required fields `traceId`, `messageNumber` and `eventTimestamp` (see description in schema)
- Removal of `latitudeDirection` and `longitudeDirection`
- Change topic to `/sensors/location`

mikaelbr committed:
docs: updates topic tree in readme

mikaelbr committed:
updates schema desc to be relevant for non-ruter

daniil-pastuhov commented:
:thumbsup: (MtBus)

NilsAndreasT commented:
MT-Buss approves.

mikaelbr commented:
@Topguy comments?

Thorbear commented:
Approved by Fara, with the new adt v3 changes.

mikaelbr commented:
Considering it approved and doing separate flow with Consat for now.
